src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -2699,6 +2781,10 @@ public void close() throws SQLServerException {
             tdsChannel.close();
         }
 
+        // Invalidate statement cache.
+        if(null != this.preparedStatementCache)
+            this.preparedStatementCache.clear();
```

Comment:
For this case the intention is luckily not to run the removal logic because all statement handles on the server side are discarded when the connection is closed. Thanks for reviewing, this is a pretty big change so definitely needs a thorough go over :-)